### PR TITLE
doc: add sanitizer test related docs

### LINKED ISSUE #4751: Add sanitizer related docs
## Enhancement
Currently, we don't have any specification on how to run ut with sanitizers.

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
### What problem does this PR solve?

Issue Number: close #4751 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -201,6 +201,26 @@ ninja gtests_libdaemon
...
+cmake $WORKSPACE/tiflash -GNinja -DENABLE_TESTS=ON -DCMAKE_BUILD_TYPE=ASan # or TSan
+ninja gtests_dbms
+ninja gtests_libcommon
+ninja gtests_libdaemon
+```
+
+There are known false positives reported from leak sanitizer (which is included in address sanitizer). To suppress these errors, set the following environment variables before running the executables:
+
+```
+LSAN_OPTIONS=suppressions=/path/to/tiflash/test/sanitize/asan.suppression
```
Comment (fuzhe1989) [inline README.md:221]: do we have any pregenerated asan.suppression file?

Comment (SchrodingerZhu) [reply README.md:221]: yes, following the path.

Comment (JaySon-Huang) [reply README.md:221]: maybe use `LSAN_OPTIONS=suppressions=$WORKSPACE/tiflash/test/sanitize/asan.suppression`

Comment (fuzhe1989) [reply README.md:221]: @JaySon-Huang that's clear enough to me.

Comment (SchrodingerZhu) [inline README.md:221]:
```suggestion
LSAN_OPTIONS=suppressions=$WORKSPACE/tiflash/test/sanitize/asan.suppression
```

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**